### PR TITLE
aws - sagemaker transform job resource

### PR DESCRIPTION
**Pull request by faizanfahmed**
Added an AWS resource type for SageMaker transform jobs. Includes basic testing and mock data.

### CODE REVIEW

File: c7n/resources/sagemaker.py [hunk 1]
```diff
@@ -107,6 +107,54 @@ def _augment(j):
...
+
+        def _augment(j):
+            tags = self.retry(client.list_tags,
+                ResourceArn=j['TransformJobArn'])['Tags']
+            j['Tags'] = tags
+            return j
+
+        jobs = super(SagemakerTransformJob, self).augment(jobs)
+        with self.executor_factory(max_workers=2) as w:
+            return universal_augment(
```
Comment (kapilt) [inline c7n/resources/sagemaker.py]: the universal augment should substitute for this augment not in addition to it.

**Comment by kapilt** [comment]
also please just push updates to the existing pr, its hard to track the conversation if every push is closing the existing discussion.

File: c7n/resources/sagemaker.py [hunk 2]
```diff
@@ -107,6 +107,55 @@ def _augment(j):
...
+
+        def _augment(j):
+            tags = self.retry(client.list_tags,
+                ResourceArn=j['TransformJobArn'])['Tags']
+            j['Tags'] = tags
+            return j
+
+        jobs = super(SagemakerTransformJob, self).augment(jobs)
+        for j in jobs:
+            j['Tags'] = _augment(j)['Tags']
```
Comment (kapilt) [inline c7n/resources/sagemaker.py]: the assignment is extraneous the augment is already mutating the dictionary.

**Comment by kapilt** [comment]
previous pr #3041

File: c7n/resources/sagemaker.py (hunk 2 above)
Comment (faizanfahmed) [reply c7n/resources/sagemaker.py]: I removed the reassignment. wrt to previous PR added a filter_name, maintaining the id and name, and switched to looping through jobs in augment.

**Pull request merged**